<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>453</width>
    <height>550</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../multimc.qrc">
    <normaloff>:/icons/toolbar/settings</normaloff>:/icons/toolbar/settings</iconset>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <item>
    <widget class="QTabWidget" name="settingsTabs">
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <widget class="QWidget" name="generalTab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="generalTabLayout">
       <item>
        <widget class="QGroupBox" name="sortingModeBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Sorting Mode</string>
         </property>
         <layout class="QHBoxLayout" name="sortingModeBoxLayout">
          <item>
           <widget class="QRadioButton" name="sortLastLaunchedBtn">
            <property name="text">
             <string>By last launched</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">sortingModeGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="sortByNameBtn">
            <property name="text">
             <string>By name</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">sortingModeGroup</string>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="updateSettingsBox">
         <property name="title">
          <string>Update Settings</string>
         </property>
         <layout class="QVBoxLayout" name="updateSettingsBoxLayout">
          <item>
           <widget class="QCheckBox" name="devBuildsCheckBox">
            <property name="text">
             <string>Use development builds?</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="autoUpdateCheckBox">
            <property name="text">
             <string>Check for updates when MultiMC starts?</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="foldersBox">
         <property name="title">
          <string>Folders</string>
         </property>
         <layout class="QGridLayout" name="foldersBoxLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="labelInstDir">
            <property name="text">
             <string>Instances:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="instDirTextBox"/>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="instDirBrowseBtn">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelModsDir">
            <property name="text">
             <string>Mods:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="modsDirTextBox"/>
          </item>
          <item row="1" column="2">
           <widget class="QToolButton" name="modsDirBrowseBtn">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelLWJGLDir">
            <property name="text">
             <string>LWJGL:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lwjglDirTextBox"/>
          </item>
          <item row="2" column="2">
           <widget class="QToolButton" name="lwjglDirBrowseBtn">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="generalTabSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="minecraftTab">
      <attribute name="title">
       <string>Minecraft</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="windowSizeGroupBox">
         <property name="title">
          <string>Window Size</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="maximizedCheckBox">
            <property name="text">
             <string>Start Minecraft maximized?</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayoutWindowSize">
            <item row="1" column="0">
             <widget class="QLabel" name="labelWindowHeight">
              <property name="text">
               <string>Window height:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="labelWindowWidth">
              <property name="text">
               <string>Window width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowWidthSpinBox">
              <property name="minimum">
               <number>854</number>
              </property>
              <property name="maximum">
               <number>65536</number>
              </property>
              <property name="singleStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>854</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="windowHeightSpinBox">
              <property name="minimum">
               <number>480</number>
              </property>
              <property name="maximum">
               <number>65536</number>
              </property>
              <property name="value">
               <number>480</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="consoleSettingsBox">
         <property name="title">
          <string>Console Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QCheckBox" name="showConsoleCheck">
            <property name="text">
             <string>Show console while the game is running?</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="autoCloseConsoleCheck">
            <property name="text">
             <string>Automatically close console when the game quits?</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="autoLoginCheckBox">
         <property name="text">
          <string>Login automatically when an instance icon is double clicked?</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacerMinecraft">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="javaTab">
      <attribute name="title">
       <string>Java</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="memoryGroupBox">
         <property name="title">
          <string>Memory</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="1">
           <widget class="QSpinBox" name="maxMemSpinBox">
            <property name="minimum">
             <number>512</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
            <property name="singleStep">
             <number>128</number>
            </property>
            <property name="value">
             <number>1024</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelMinMem">
            <property name="text">
             <string>Minimum memory allocation:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelMaxMem">
            <property name="text">
             <string>Maximum memory allocation:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="minMemSpinBox">
            <property name="minimum">
             <number>256</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
            <property name="singleStep">
             <number>128</number>
            </property>
            <property name="value">
             <number>256</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelPermGen">
            <property name="text">
             <string>PermGen:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="permGenSpinBox">
            <property name="minimum">
             <number>64</number>
            </property>
            <property name="maximum">
             <number>512</number>
            </property>
            <property name="singleStep">
             <number>8</number>
            </property>
            <property name="value">
             <number>64</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="javaSettingsGroupBox">
         <property name="title">
          <string>Java Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="labelJavaPath">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Java path:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelJVMArgs">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>JVM arguments:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="btnBrowse">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QLineEdit" name="javaPathTextBox"/>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="pushButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Auto-detect</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="3">
           <widget class="QLineEdit" name="jvmArgsTextBox"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="customCommandsGroupBox">
         <property name="title">
          <string>Custom Commands</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="1" column="0">
           <widget class="QLabel" name="labelPostExitCmd">
            <property name="text">
             <string>Post-exit command:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelPreLaunchCmd">
            <property name="text">
             <string>Pre-launch command:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="preLaunchCmdTextBox"/>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="postExitCmdTextBox"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelCustomCmdsDescription">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Pre-launch command runs before the instance launches and post-exit command runs after it exits. Both will be run in MultiMC's working directory with INST_ID, INST_DIR, and INST_NAME as environment variables.</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>settingsTabs</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>sortLastLaunchedBtn</tabstop>
  <tabstop>sortByNameBtn</tabstop>
  <tabstop>devBuildsCheckBox</tabstop>
  <tabstop>autoUpdateCheckBox</tabstop>
  <tabstop>instDirTextBox</tabstop>
  <tabstop>modsDirTextBox</tabstop>
  <tabstop>lwjglDirTextBox</tabstop>
  <tabstop>instDirBrowseBtn</tabstop>
  <tabstop>modsDirBrowseBtn</tabstop>
  <tabstop>lwjglDirBrowseBtn</tabstop>
  <tabstop>maximizedCheckBox</tabstop>
  <tabstop>windowWidthSpinBox</tabstop>
  <tabstop>windowHeightSpinBox</tabstop>
  <tabstop>showConsoleCheck</tabstop>
  <tabstop>autoCloseConsoleCheck</tabstop>
  <tabstop>autoLoginCheckBox</tabstop>
  <tabstop>minMemSpinBox</tabstop>
  <tabstop>maxMemSpinBox</tabstop>
  <tabstop>permGenSpinBox</tabstop>
  <tabstop>javaPathTextBox</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>btnBrowse</tabstop>
  <tabstop>jvmArgsTextBox</tabstop>
  <tabstop>preLaunchCmdTextBox</tabstop>
  <tabstop>postExitCmdTextBox</tabstop>
 </tabstops>
 <resources>
  <include location="../multimc.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="sortingModeGroup"/>
 </buttongroups>
</ui>
